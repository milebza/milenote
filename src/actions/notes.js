import uuid from 'uuid'

// ADD_NOTE
export const addNote = ({ title = '', content = '',  date = 0 } = {}) => ({
  type: 'ADD_NOTE',
  Note: {
    id: uuid(),
    title,
    content,
    date
  }
})

// REMOVE_NOTE
export const removeNote = ({ id } = {}) => ({
  type: 'REMOVE_NOTE',
  id
})

// EDIT_NOTE
export const editNote = (id, updates) => ({
  type: 'EDIT_NOTE',
  id,
  updates
})
