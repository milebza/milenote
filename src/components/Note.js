import React from 'react';

const Note = ({title, content}) => (
  <div className="form padding-075">
    <h3 className="note-info__title">{title}</h3>
    <p className="note-info__p">{content}</p>
  </div>
)

export default Note
