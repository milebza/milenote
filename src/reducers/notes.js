// Notes Reducer
const NotesReducerDefaultState = []

export default (state = NotesReducerDefaultState, action) => {
  switch (action.type) {
    case 'ADD_NOTE':
      return [
        ...state,
        action.note
      ]
    case 'REMOVE_NOTE':
        return state.filter(({ id }) => (id !== action.id))
    case 'EDIT_NOTE':
      return state.map((note) => {
        if (note.id === action.id) {
          return {
            ...note,
            ...action.updates
          }
        }
        return note
      })
    case 'SET_NOTES':
        return action.notes
    default:
      return state
  }
}
