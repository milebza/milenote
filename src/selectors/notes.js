import moment from 'moment'

// Get visible Notes
export default (notes, {text, startDate, endDate}) => {
  return notes.filter((note) => {
    const date = moment(note.date)
    const startDateMatch = startDate ? startDate.isSameOrBefore(date, 'day') : true
    const endDateMatch = endDate ? endDate.isSameOrAfter(date, 'day') : true
    const textMatch = note.title.toLowerCase().includes(text.toLowerCase()) || note.content.toLowerCase().includes(text.toLowerCase())

    return startDateMatch && endDateMatch && textMatch
  }).sort((a, b) => {
    // 1 -> b will go first
    return a.date < b.date ? 1 : -1
  })
}
