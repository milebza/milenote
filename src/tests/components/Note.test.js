import React from 'react'
import { shallow } from 'enzyme'
import Note from '../../components/Note'
import notes from '../fixtures/notes'

test('should render Note with note content', () => {
  const wrapper = shallow(<Note {...notes[0]} />)
  expect(wrapper).toMatchSnapshot()
})
