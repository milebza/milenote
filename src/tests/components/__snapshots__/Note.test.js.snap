// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Note with note content 1`] = `
<div
  className="form"
>
  <div
    className="form__actions"
  >
    <button
      className="btn-link form__btn form__btn-edit"
    >
      <span
        className="icon icon-pencil"
      />
       Edit
    </button>
  </div>
  <hr
    className="mobile-show"
  />
  <div
    className="note-info__wrapper"
  >
    <h3
      className="note-info__title"
    >
      Letter
    </h3>
    <textarea
      className="form__input note-info__content"
      disabled={true}
      rows="20"
      value="Dear friend"
    />
  </div>
</div>
`;
